collections/pick_test.ts

```diff
@@ -37,3 +50,18 @@ Deno.test({
     assertNotStrictEquals(picked, obj);
   },
 });
+
+Deno.test({
+  name:
+    "pick() returns a new object with own properties that is non-own in the original object)",
+  fn() {
+    // deno-lint-ignore no-explicit-any
+    const obj = { a: 5, b: 6, c: 7, d: 8 } as any;
+    const picked: { toString: typeof Object.prototype.toString } = pick(obj, [
+      "toString",
+    ]);
+
+    assertEquals(picked, { toString: Object.prototype.toString });
```

Comment:
I personally don't think this is intentionally in this way.. (The check with `Object.hasOwn` meets the users' expectation better in my view)

@lambdalisue What do you think?

Reply:
Considering the original purpose, I hadn't envisioned a use case where `toString` would be picked. Therefore, I think @kt3k's suggestion is a good one. However, since the implementation currently references `obj[k]`, changing it to `Object.hasOwn` would be a **breaking change**. Additionally, in terms of type expression, it's written as `K in keyof T`, which means it allows something like `toString` (the following code does not result in a type error).

```ts
import { pick } from "jsr:@std/collections/pick";
import { omit } from "jsr:@std/collections/omit";

class A {
  constructor(public a: string, public b: string, public c: string) {}

  foo(): string {
    return "hello";
  }
}

const obj = new A("a", "b", "c");

const p = pick(obj, ["a", "c", "foo"]);
const o = omit(obj, ["b"]);

console.log(obj.foo());
console.log(p.foo());
console.log(o.foo());
```

This might be beyond the scope of this PR, but I realized that `o.foo()` does not work in the code above. Thus, currently there is an inconsistency between `pick` and `omit`. Given this situation, we need to choose between the following options:

1. Restrict the targets of `pick` and `omit` to `Object.hasOwn` (and adjust the type so it expresses this correctly).
2. Fix the implementation of `omit` so that `o.foo()` does not result in a runtime error.

Personally, I prefer option 1, but I’m not sure how to express this with types.

Reply:
Ah, ok. The current check (`key in obj`) works good for the case of the above `class A` example (`foo` method should be picked in that case).

I'm now in favor of landing this fix.

However, I'm still not sure we should include this particular example (`pick(obj, ["toString"])`) as a test case, as this causes type error if we don't type cast `obj` to any. I think we can consider this particular behavior something like undefined/not decided behavior.

Reply:
How about just removing this test case and revisit later when someone comes up with something better?

Reply:
I'm not opposed to it, but just curious, is the concept of “undefined behavior” common in the standard library? If we decide to make this UB, there has to be an explicit notice in the JSDoc comment.

Reply:
It's not common to make something explicitly undefined behavior in the standard library, but I think we are not ready to decide about this behavior, and also this looks out of scope of this PR.

In my view, in the below example, `pick`ing `foo` looks fine, but `pick`ing `toString` looks strange as it doesn't match its type (it's actually causes a type error).

```ts
import { pick } from "@std/collections";
class A {
  foo() {
  }
  bar() {
  }
}
const a = new A();
pick(a, ["foo"]);
pick(a, ["toString"]);
```